client/lib/explat/internals.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * WordPress dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+
+/**
+ * Internal dependecies
+ */
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { getAnonIdFromCookie } from 'calypso/state/experiments/reducer';
+
+// Using typescript to ensure we are being safe in SSR
+declare const window: undefined | ( Window & typeof globalThis );
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+declare const document: undefined | Document;
+
+export const isDevelopmentMode = process.env.NODE_ENV === 'development';
+
+/**
+ * Log an error to the server.
+ * Works in SSR.
+ *
+ * @param error Error to save
+ */
+export const logError = ( error: Record< string, string > & { message: string } ): void => {
+	try {
+		if ( isDevelopmentMode ) {
+			// eslint-disable-next-line no-console
+			console.error( '[ExPlat]', error.message, error );
+		}
+
+		error[ 'exPlatStandaloneClient' ] = 'calypso';
+
+		if ( typeof window === 'undefined' ) {
+			// We console.log in SSR contexts, this get's sent to logStash
+			// eslint-disable-next-line no-console
+			console.log( '[ExPlat]', error.message, error );
+		} else {
+			const body = new window.FormData();
+			body.append( 'error', JSON.stringify( error ) );
+
+			// TODO: Use wp.req.post here too
+			window.fetch( 'https://public-api.wordpress.com/rest/v1.1/js-error', {
+				method: 'POST',
+				body,
+			} );
+		}
+	} catch {
+		if ( isDevelopmentMode ) {
+			// eslint-disable-next-line no-console
+			console.error( '[ExPlat] Unable to send error to server.' );
+		}
+	}
+};
+
+export const fetchExperimentAssignment = ( {
+	experimentName,
+	anonId,
+}: {
+	experimentName: string;
+	anonId?: string;
+} ): Promise< unknown > => {
+	return wpcom.req.get(
```

Comment:
@jsnajdr is it ok to use `wpcom.req.get`?

Reply:
My $0.02 is that while you can use it here, it's indeed outdated and obsolete. We still do have a bunch of `wpcom` lib usage, but it usually is wrapped in old `wpcom` methods and we've been migrating away from it when that makes sense. A big downside when using `wpcom` is that it's monolithic and you have to pull all of it in your project, just for a simple request. Another one is that you don't really have a huge control over the request itself. I'd recommend using the data layer here - or because this is such a simple scenario, you can just introduce a new data getter under `client/state/data-getters`. Let me know if that makes sense.